File: _REPO/APRESS/distributed-.net-programming-in-vb-.net/Chapter05/RemotingMetadata/CustomerLibrary/CAOCustomer.vb
Public Class CAOCustomer
   Inherits MarshalByRefObject
   
   Private mName As String

   Public Sub New(name As String)
      Console.WriteLine("Customer.ctor(""{0}"")", name)
      mName = name
   End Sub
   
   Public Function SayHello() As String
      Console.WriteLine("Customer.SayHello")
      Return "Hello " & mName
   End Function
End Class

File: _REPO/APRESS/distributed-.net-programming-in-csharp/Chapter05/RemotingMetadata/CustomerLibrary/CAOCustomer.cs
using System;

namespace CustomerLibrary
{
   public class CAOCustomer : MarshalByRefObject
   {
      string mName;

      public CAOCustomer(string name)
      {
         Console.WriteLine("Customer.ctor(\"{0}\")", name);
         mName = name;
      }
      public string SayHello()
      {
         Console.WriteLine("Customer.SayHello");
         return "Hello " + mName;
      }
   }
}
